frontend/lib/resolution/scope-types.cpp

```diff
@@ -42,6 +42,86 @@ std::string IdAndFlags::flagsToString(Flags flags) {
   return ret;
 }
 
+using Flags = IdAndFlags::Flags;
+using FlagSet = IdAndFlags::FlagSet;
+
+FlagSet FlagSet::singleton(Flags flags) {
+  FlagSet toReturn;
+  toReturn.insert(flags);
+  return toReturn;
+}
+
+FlagSet FlagSet::empty() {
+  return FlagSet();
+}
+
+void FlagSet::insert(Flags excludeFlags) {
+  // booleans, like all lattices, follow the absorption law:
+  //
+  //     a /\ (a \/ b) = a   and   a \/ (a /\ b) = a
```

Comment:
AFAIK we can have UTF-8 in comments and I think this would be easier to read that way, but it's not a big deal and probably not worth changing